gomint-api/src/main/java/io/gomint/event/world/SignChangeTextEvent.java

```diff
@@ -49,38 +63,58 @@ public Block getBlock() {
      * @param line    which should be set
      * @param content which should be set on that line
      */
-    public void setLine( int line, String content ) {
+    public void setLine(int line, String content) {
         // Silently fail when line is incorrect
-        if ( line > 4 || line < 1 ) {
+        if (line > 4 || line < 1) {
             return;
         }
 
-        if ( this.lines.size() < line ) {
-            for ( int i = 0; i < line - this.lines.size(); i++ ) {
-                this.lines.add( "" );
+        if (this.lines.size() < line) {
+            for (int i = 0; i < line - this.lines.size(); i++) {
+                this.lines.add("");
             }
         }
 
-        this.lines.set( line - 1, content );
+        this.lines.set(line - 1, content);
     }
 
     /**
      * Get a specific line of the sign content
      *
      * @param line which you want to get
-     * @return string or null when not setö
+     * @return string or null when not set
      */
-    public String getLine( int line ) {
+    public String getLine(int line) {
         // Silently fail when line is incorrect
-        if ( line > 4 || line < 1 ) {
+        if (line > 4 || line < 1) {
             return null;
         }
 
-        if ( this.lines.size() < line ) {
+        if (this.lines.size() < line) {
             return null;
         }
 
-        return this.lines.get( line - 1 );
+        return this.lines.get(line - 1);
+    }
+
+    /**
+     * Get all lines which the sign previously had
+     *
+     * @return all lines which the sign previously had
+     */
+    public List<String> getLines() {
+        return this.lines;
```

Comment:
I'm still not happy with this getter not returning a copy or unmodifyable view.

Reply:
Same, maybe a copy into a unmodifiable would be good here

Reply:
In honor of the jvm we are not creating a copy

Reply:
Well the event is likely not hot code (it doesn't trigger that often) and its more important to guarantee that a read operation does not result in a writing change

Reply:
Yea but i think it's not needed. The result will be discareded later